cliffordlayers/cliffordalgebra.py

```diff
@@ -0,0 +1,62 @@
+import torch
+
+from .basisbladeorder import ShortLexBasisBladeOrder, construct_gmt
+
+
+class CliffordAlgebra:
+    def __init__(self, metric):
+        self.metric = torch.as_tensor(metric, dtype=torch.float32)
+        self.num_bases = len(metric)
+        self.bbo = ShortLexBasisBladeOrder(self.num_bases)
+        self.dim = len(self.metric)
+        self.n_blades = len(self.bbo.grades)
+        self.cayley = construct_gmt(
+            self.bbo.index_to_bitmap, self.bbo.bitmap_to_index, self.metric
+        ).to_dense()
+
+    def to(self, device: torch.device):
+        self.bbo.grades = self.bbo.grades.to(device)
+        self.metric = self.metric.to(device)
+        self.cayley = self.cayley.to(device)
+        return self
+
+    def geometric_product(self, a, b):
+        return torch.einsum(
+            "bfi,ijk,bfk->bfj",
+            a.view(len(a), -1, len(self.cayley)),
+            self.cayley,
+            b.view(len(b), -1, len(self.cayley)),
+        ).view(*a.size())
+
+    def reverse(self, mv, blades=None):
+        grades = self.bbo.grades.to(mv.device)
```

Comment:
Would be super useful to add documentation to these functions of this class! `reverse`, `embed` are fairly generic terms and readers/users may not know what they mean in this context.

Reply:
Added these!